### DISCUSSION TITLE
Add installation step for debian 11?

### DISCUSSION THREAD

**Discussion by KisaragiEffective** [mono/mono#21574] [General]
I saw https://www.mono-project.com/download/stable/#download-lin-debian for install to my desktop. However, there's no paragraph for Debian 11. `echo "deb https://download.mono-project.com/repo/debian stable-bullseye main" | sudo tee /etc/apt/sources.list.d/mono-official-stable.list` does not work...

However, I can use toolchain without problem - so, I'd like to suggest add steps for debian 11.

**Comment by Ghostbird** [comment]
You know what's even more baffling, there are lot's of blogs that show how to install mono on Debian 11 (Bullseye) that reference `deb https://download.mono-project.com/repo/debian stable-bullseye main` as sources line. Even though that repository doesn't exist. Did it exist at some point, but was it removed?

**Reply by zocker-160** [reply]
> how to install mono on Debian 11 (Bullseye) that reference deb https://download.mono-project.com/repo/debian stable-buster main as sources line

yes that is the Debian 10 repo (see the `buster` keyword)

> Even though that repository doesn't exist. Did it exist at some point, but was it removed?

The Debian 10 buster repository does still exist and works on Debian 11 as well. A Debian 11 bullseye repository did never exist.

**Reply by Ghostbird** [reply]
Sorry, my mistake, I copied the line, from a correct source, and meant to change `buster` to `bullseye`. I've fixed my comment to reflect that.
My point was that it seems several blogs, simply take the instructions for buster, change buster to bullseye, and publish that as if it's the truth.

**Reply by Ghostbird** [reply]
But I do think it would be nice if there was an official bullseye repository.

**Reply by zocker-160** [reply]
> Sorry, my mistake, I copied the line, from a correct source, and meant to change buster to bullseye. I've fixed my comment to reflect that.
My point was that it seems several blogs, simply take the instructions for buster, change buster to bullseye, and publish that as if it's the truth.

Ah I understand now, but then those blogs were posting nonsense instructions, since IIRC the bullseye repository never existed.

Yes having proper bullseye repositories would be great, but given that they haven't updated any other repository either (still only centOS 7 and 8, no 9 and still only Ubuntu focal and no jammy), I kinda doubt this will happen.

**Reply by Ghostbird** [reply]
Thanks, I thought the repo never existed, but I wasn't sure, since it seemed strange to me that those blogs would post faulty instructions for something easily verifiable.

Interesting, I didn't know that other Linux distro's also had no updated repositories. I thought it was an oversight, and a happy coincidence that the buster repo worked. But in this light, it seems more like a deliberate strategy. Probably with a sensible reason, even if I don't know it.

**Reply by zocker-160** [reply]
Out of curiosity I did a bit of research:

- https://vitux.com/how-to-install-and-use-mono-on-debian-11/ (invalid url)
- https://itslinuxfoss.com/how-to-install-mono-on-debian-11/ (invalid url - the screenshots make me suggest they did not even actually test it lol)
- https://www.howtoforge.com/how-to-install-mono-on-debian-11/ (using buster repo)
- https://www.linuxcapable.com/how-to-install-mono-framework-on-debian-11-bullseye/ (using buster repo)
- https://www.techlear.com/blog/2021/11/01/how-to-install-mono-on-debian-11/ (using buster repo)
- https://www.linuxcompatible.org/story/how-to-install-and-use-mono-on-debian-11/ (linking to vitux which uses invalid url)
